crates/iceberg/src/spec/view_version.rs

```diff
@@ -0,0 +1,347 @@
+    }
+
+    /// Get the schema id of this view version.
+    #[inline]
+    pub fn schema_id(&self) -> SchemaId {
+        self.schema_id
+    }
+
+    /// Get the timestamp of when the view version was created
+    #[inline]
+    pub fn timestamp(&self) -> MappedLocalTime<DateTime<Utc>> {
+        Utc.timestamp_millis_opt(self.timestamp_ms)
+    }
+
+    /// Get the timestamp of when the view version was created in milliseconds since epoch
+    #[inline]
+    pub fn timestamp_ms(&self) -> i64 {
+        self.timestamp_ms
+    }
+
+    /// Get summary of the view version
+    #[inline]
+    pub fn summary(&self) -> &HashMap<String, String> {
+        &self.summary
+    }
+
+    /// Get this views representations
+    #[inline]
+    pub fn representations(&self) -> &ViewRepresentations {
+        &self.representations
+    }
+
+    /// Get the default catalog for this view version
+    #[inline]
+    pub fn default_catalog(&self) -> Option<&String> {
+        self.default_catalog.as_ref()
+    }
+
+    /// Get the default namespace to use when a reference in the SELECT is a single identifier
+    #[inline]
+    pub fn default_namespace(&self) -> &NamespaceIdent {
+        &self.default_namespace
+    }
+
+    /// Get the schema of this snapshot.
+    pub fn schema(&self, view_metadata: &ViewMetadata) -> Result<SchemaRef> {
+        let r = view_metadata
+            .schema_by_id(self.schema_id())
+            .ok_or_else(|| {
+                Error::new(
+                    ErrorKind::DataInvalid,
+                    format!("Schema with id {} not found", self.schema_id()),
+                )
+            })
+            .cloned();
+        r
+    }
+
+    pub(crate) fn log(&self) -> ViewVersionLog {
+        ViewVersionLog::new(self.version_id, self.timestamp_ms)
+    }
+}
+
+/// A list of view representations.
+pub type ViewRepresentations = Vec<ViewRepresentation>;
+
+/// A builder for [`ViewRepresentations`].
+pub struct ViewRepresentationsBuilder(ViewRepresentations);
+
+impl ViewRepresentationsBuilder {
+    /// Create a new builder.
+    pub fn new() -> Self {
+        Self(Vec::new())
+    }
+
+    /// Add a representation to the list.
+    ///
+    /// SQL representations dialects must be unique. If a representation with the same
+    /// dialect already exists, it will be overwritten.
+    pub fn add_or_overwrite_representation(mut self, representation: ViewRepresentation) -> Self {
+        let dialect = match &representation {
+            ViewRepresentation::SqlViewRepresentation(sql) => &sql.dialect,
+        };
+        self.0.retain(|r| {
+            let ViewRepresentation::SqlViewRepresentation(sql) = r;
+            sql.dialect != *dialect
+        });
+        self.0.push(representation);
+        self
+    }
+
+    /// Add a SQL representation to the list.
+    ///
+    /// SQL representations dialects must be unique. If a representation with the same
+    /// dialect already exists, it will be overwritten.
```

Comment:
why is it better to overwrite a dialect rather than throw an error saying that there are duplicate dialects?

Reply:
I guess it depends on the usage. This method is part of a builder which is not only used to create an object initially, but also to modify an object.
If we fail, there is no way to ever modify a dialect in the builder.
I am adding fallible methods additionally in https://github.com/c-thiel/iceberg-rust/commit/83e4a6f516cc064e7949d2b48ccd615a3b3d0c3c (tests in next commit)